src/utilities/useAxios.ts

```diff
@@ -0,0 +1,54 @@
+import { useAuth0 } from "@auth0/auth0-react";
+import axios from "axios";
+import { useCallback } from "react";
+
+export type SendRequestProps = {
+  endpoint: string;
+  method?: "GET" | "POST" | "PUT" | "DELETE";
+  headers?: { [key: string]: string };
+  body?: any;
+  params?: any;
+};
+
+const useAxios = () => {
+  const { getAccessTokenSilently } = useAuth0();
+
+  const sendRequest = useCallback(
```

Comment:
I kinda kept this here as a precaution because I think it is good practice to have a useCallback on a function that runs in a useEffect. It could lead to unneccessary re-renders, so I was thinking we can memoize the function to avoid that